gradio/components/chatbot.py

```diff
@@ -37,6 +37,8 @@ class MetadataDict(TypedDict):
     title: Union[str, None]
     id: NotRequired[int | str]
     parent_id: NotRequired[int | str]
+    duration: NotRequired[int]
+    status: NotRequired[Literal["pending", "done"]]
```

Comment:
Note that I moved `duration` inside of `metadata` where I think it makes more sense. I also added a `status` for more control over the loading spinner, as it improves the UI a bit (cc @yvrjsharma)